guides/hack/02-source-code-fundamentals/37-namespaces.md

```diff
@@ -1,30 +1,39 @@
-A problem encountered when managing large projects is that of avoiding the use of the same name in the same scope for different
-purposes. This is especially problematic in a language that supports modular design and component libraries.  This problem is
-addressed by namespaces.
+A ***namespace*** is a container for a set of (typically related) definitions of classes, interfaces, traits, functions, and constants.
 
-A *namespace* is a container for a set of (typically related) definitions of classes, interfaces, traits, functions, and constants.
+When the same namespace is defined in multiple scripts, and those scripts are combined into the same program, the namespace
+is the union of all individual components.
 
-A namespace may have *sub-namespaces*, where a sub-namespace name shares a common prefix with another namespace. For example, the
-namespace `Graphics` might have sub-namespaces `Graphics\TwoD` and `Graphics\ThreeD`, for two- and three-dimensional facilities,
-respectively. Apart from their common prefix, a namespace and its sub-namespaces have no special relationship. The namespace
-whose prefix is part of a sub-namespace need not actually exist for the sub-namespace to exist. That is, `NS1\Sub` can exist
-without `NS1`.
+The predefined constant `__NAMESPACE__` contains the name of the current namespace.
 
+## The Root Namespace
 In the absence of any namespace definition, the names of subsequent classes, interfaces, traits, functions, and constants are in
-the *default namespace*, which has no name, per se.
+the ***root namespace***, which is not named.
+
+## Sub-Namespaces
+A namespace can have ***sub-namespaces***, where a sub-namespace name shares a common prefix with another namespace. 
+
+For example, the namespace `Graphics` can have sub-namespaces `Graphics\TwoD` and `Graphics\ThreeD`, for two- and three-dimensional facilities,
+respectively. 
 
-The names of some types (such as `Exception`), constants, and library functions (such as `sqrt`) are defined outside any namespace.
-To refer unambiguously to such names, one can prefix them with a backslash (`\`), as in `\Exception` and `\sqrt`. The names of
-the standard types that are introduced with Hack belong to namespace `HH`.
+Apart from their common prefix, a namespace and its sub-namespaces have no special relationship. For example, `NS1\Sub` can exist without `NS1`. A named top-level namespace does not need to exist for a sub-namespace to exist.
 
+## Reserved Namespaces
 The namespaces `HH`, `PHP`, `php`, and sub-namespaces beginning with those prefixes are reserved for use by Hack.
 
-The pre-defined constant `__NAMESPACE__` contains the name of the current namespace.
+### The `HH` Namespace
+Built-in types like `vec`, `keyset`, `dict`, and `shape` exist in the `HH` namespace (e.g. `HH\Lib\Vec`).
```

Comment:
The example provided is misconception. The fully qualified name for `vec<T>` is `\HH\vec<T>`. The name vec is automatically imported into all Hack files. The `\HH\Lib\Vec` namespace is related, but not where `vec<T>` is declared.

Reply:
Apologies, thank you.